### PR TITLE
For #77: Errors page and tests.

### PR DESCRIPTION
**Pull request by paulodamaso**
For #77:
- Created error take and page;
- Added tests;
- Added fake classes for testing
- Left puzzle for error delete implementation

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#80](https://github.com/yegor256/wring/issues/80) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#80](https://github.com/yegor256/wring/issues/80) is assigned to @pinaf[/z](https://www.0crat.com/u/pinaf), here is [why](https://www.0crat.com/footprint/C7FCB1EQN/2ca497ed-4e3c-46d2-9003-5fd6360fdd9c); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by paulodamaso** [comment]
@pinaf ping

File: src/main/java/io/wring/tk/TkErrors.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+            new XeAppend(
+                "errors",
+                new XeChain(
+                    new XeTransform<>(
+                        Iterables.limit(errors, Tv.TWENTY),
+                        TkErrors::source
+                    )
+                ),
+                new XeDirectives(
+                    new Directives().attr("total", Iterables.size(errors))
```
Comment (pinaf) [inline src/main/java/io/wring/tk/TkErrors.java]: @paulodamaso are we sure we can iterate over `errors` twice without any issues?

File: src/test/java/io/wring/tk/TkErrorsTest.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+                        )
+                    )
+                ).printBody()
+            ),
+            XhtmlMatchers.hasXPaths(
+                "/page/millis",
+                "/page/errors/error",
+                "/page/errors/@total"
+            )
+        );
```
Comment (pinaf) [inline src/test/java/io/wring/tk/TkErrorsTest.java]: @paulodamaso let's expand this test with a few fake errors so we can check the resulting xml for them.

File: src/main/java/io/wring/fake/FkErrors.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import java.util.Collections;
+
+/**
+ * Fake errors.
+ *
+ * @author Paulo Lobo (pauloeduardolobo@gmail.com)
+ * @version $Id$
+ * @since 1.0
+ */
+public final class FkErrors implements Errors {
```
Comment (pinaf) [inline src/main/java/io/wring/fake/FkErrors.java]: @paulodamaso let's allow for a custom list of errors to be set via a constructor for better unit testing.

File: src/test/java/io/wring/tk/TkErrorsTest.java (hunk 2 above)
Comment (paulodamaso) [reply src/test/java/io/wring/tk/TkErrorsTest.java]: @pinaf we already have how to check the resulting xml, using FkError.

Comment (pinaf) [reply src/test/java/io/wring/tk/TkErrorsTest.java]: @paulodamaso you are only checking for the presence of some xpaths. I'm proposing to check that those xpaths match the error's contents.

**Comment by paulodamaso** [comment]
@pinaf Please take a look. I've decided to use `FkError` for testing the page because implementing it to accept values would become too close to `Error.Simple` implementation made in https://github.com/yegor256/wring/issues/76, and I don't want to duplicate this code.

**Comment by pinaf** [comment]
@paulodamaso thanks, it looks good.
@carlosmiranda good to merge.

**Comment by carlosmiranda** [comment]
@rultor merge

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @pinaf[/z](https://www.0crat.com/u/pinaf), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#80](https://github.com/yegor256/wring/issues/80) is now out of scope

**Comment by ypshenychka** [comment]
@pinaf According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Only two issues were found during code review. 
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by ypshenychka** [comment]
@pinaf thanks

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 point(s) just awarded to @pinaf[/z](https://www.0crat.com/u/pinaf)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**